### DISCUSSION TITLE
HTTP 500 Server Error returned with intentionally empty tiles (aka missing tiles), ideally should return HTTP 204 No Content

### DISCUSSION THREAD

**Discussion by davidcalhoun** [maplibre/martin#1188] [General]
Hi folks, thanks for making this great library!

I'm running into a situation where our UI makes requests for areas where there's no data (empty tiles).  In these cases `martin` returns HTTP 500 responses which causes a lot of spam in our network logs (too many false positive errors, falsely indicating there's a problem with the server).  In this situation the empty tile is normal and expected(not a server error as implied by HTTP 500) and IMO should be something nominal like HTTP 204 No Content.  Is there any way to configure this?

There's suggestions online to alternatively fill all the empty tiles with stub data or something, but this seems not ideal and wasteful in terms of added size.  I'm open to any other alternatives though.

Thank you in advance!

**Comment by davidcalhoun** [comment]
Extra context: [mbtileserver](https://github.com/consbio/mbtileserver) returns HTTP 204 No Content for missing tiles, which is more ideal, see their [readme](https://github.com/consbio/mbtileserver/tree/04d08f1258e1183c55cf8f5d1fb39015fff73aab?tab=readme-ov-file#missing-tiles).

**Reply by nyurik** [reply]
Martin should not return 500 if a tile is missing, and has special handling for that.  If this is failing, it means the specific backend you use (pg/mbtiles/pmtiles) has incorrect implementation, and returns an error on empty - I will need to reproduce this to fix it.  Please provide martin version, and all the specifics of your backend. Thx!

https://github.com/maplibre/martin/blob/7be4360b7c751a012770187a9514b3602bc99c06/martin/src/srv/tiles.rs#L104-L105

**Reply by davidcalhoun** [reply]
Thank you for the quick reply!  I'll take a look and provide more info when I can.

**Reply by davidcalhoun** [reply]
Thanks again for the input, I know it's been some months.  I'm looking into this more and it seems likely that the HTTP 204 is actually getting converted to HTTP 500 by our GCP load balancer, but I still need to confirm.  Just wanted to provide this info in case folks experience similar issues.

**Reply by davidcalhoun** [reply]
Haven't had time to look into this more but the current suspicion is that there's something weird in the way we're building our tiles and outputting pmtiles.  We seemed to start getting HTTP 500 after switching from using mbtiles to pmtiles.

**Reply by nyurik** [reply]
pmtiles support is relatively new, so it is quiet possible there is a bug in there somewhere

**Comment by davidcalhoun** [comment]
Closing this out, we figured out the issue!  This was fixed by upgrading from martin `v0.8.7` to `v0.13.0`.

Summary:
* `martin@v0.8.7`: empty tiles in pmtiles return HTTP 500
* `martin@v0.13.0`: empty tiles in pmtiles return HTTP 204 (as expected and desired!)

**Reply by nyurik** [reply]
glad to hear it, thx.  Btw v0.14 was just released
